Fix macOS building, CMake policies and finding Boost behaviour (#472)
Desbordante/desbordante-core: README.md

#### MacOS dependencies installation

Install Xcode Command Line Tools if you don't have them. Run:
```sh
xcode-select --install
```

> Follow the prompts to contunue.

Reviewer: ```suggestion
Follow the prompts to continue.
```

> You'll see a panel that asks you to install Xcode Command Line Tools. Click 'Install' to begin the installation process and wait for it to complete.

Reviewer 2: IMHO this feels like spoon-feeding.
I feel like
```md
#### Install Xcode Command Line Tools

Open Terminal and run:

```sh
xcode-select --install

Follow the prompts to install.
```

is sufficient

To install GCC and CMake on macOS we recommend to use [Homebrew](https://brew.sh/) package manager. With Homebrew
installed, run the following commands:
```sh

> brew install gcc@14 cmake 

Reviewer: remove trailing spaces (they are also on the lines 263, 280, 282, 283)

Reviewer: Seems, like this line wasn't updated

Author: Fixed

```
After installation, check `cmake --version`. If command is not found, then you need to add to environment path to
homebrew installed packages. To do this open `~/.zprofile` (for Zsh) or
`~/.bash_profile` (for Bash) and add to the end of the file the output of `brew shellenv`.
After that, restart the terminal and check the version of CMake again, now it should be displayed.

Then you need to install Boost library built with GCC. Please, don't use Homebrew for this, because brew boost is built with Clang.

> Instead, download the latest version of Boost from the [official website](https://www.boost.org/users/download/).

Reviewer 2: And why is it bad that brew boost is built with clang? This needs some explanation
I.e 

> Please avoid using Homebrew for this, as the Boost version provided by Homebrew is built with Clang, which..

> After downloading, unpack the archive to the `/usr/local` directory or another directory of your choice.

Reviewer: Since we are providing a very detailed guide, then perhaps we can also provide commands for downloading the boost (for example, use the current latest version of boost and the path `/usr/local` in the commands).

I would say that I would be inclined to do so, but I would not insist.

> Go to unpacked boost directory in the terminal and run the following commands:

Reviewer: In this phrase the wording could be slightly more formal:
```suggestion
Navigate to the unpacked Boost directory in the terminal and run the following commands:
```
Also we should use `the` here and use `Boost` instead of `boost` for consistency

```sh
./bootstrap.sh 
echo "using gcc : : g++-14 ;" > user-config.jam
./b2 --user-config=user-config.jam
export BOOST_ROOT=$(pwd)
``` 

> You can also add last export with current path to `~/.zprofile` or `~/.bash_profile` to set this boost path by default.

Reviewer: here we can also change `boost` to `Boost` for consistency

Before building project you must set localy or in the above mentioned files the following CMake environment variables:

> ```sh

Reviewer 2: above-mentioned

Reviewer 2: before building _the_ project

export CC=gcc-14
export CXX=g++-14
export SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/   
```
The first two lines set GCC as default compiler in CMake. The last export is needed also because of issues with GCC 14 and 
the last MacOSX15.0.sdk used by CMake by default, you can read about this [here](https://gist.github.com/scivision/d69faebbc56da9714798087b56de925a) 

> and [here](https://github.com/iains/gcc-14-branch/issues/11).

Reviewer 2: The last export is also necessary due to issues
you can read _more_ about this

homebrew installed packages. To do this open `~/.zprofile` (for Zsh) or
`~/.bash_profile` (for Bash) and add to the end of the file the output of `brew shellenv`.
After that, restart the terminal and check the version of CMake again, now it should be displayed.

Then you need to install Boost library built with GCC. Please avoid using Homebrew for this, as the Boost version provided by Homebrew
is built with Clang, which has a different ABI. Instead, download the latest version of Boost from the [official website](https://www.boost.org/users/download/) and unpack the archive to the `/usr/local/lib` directory or another directory of your choice:
```sh
cd /usr/local/lib
curl https://archives.boost.io/release/1.86.0/source/boost_1_86_0.tar.bz2 --output "boost_1_86_0.tar.bz2"
tar xvjf boost_1_86_0.tar.bz2
rm boost_1_86_0.tar.bz2
cd boost_1_86_0
```
Navigate to the unpacked Boost directory in the terminal and run the following commands:
```sh
./bootstrap.sh 
echo "using gcc : : g++-14 ;" > user-config.jam
./b2 install --user-config=user-config.jam --layout=versioned
export BOOST_ROOT=$(pwd) # export Boost_ROOT=$(pwd) for CMake 3.26 and below.
export BOOST_COMPILER=gcc14
``` 

> You can also add last 2 export with fixed Boost root path to `~/.zprofile` or `~/.bash_profile` to set them by default.

Reviewer: ```suggestion
You can also add the last two exports with the fixed Boost root path to `~/.zprofile` or `~/.bash_profile` to set them by default.
```

Before building the project you must set locally or in the above-mentioned dotfiles the following CMake environment variables:
```sh
export CC=gcc-14
export CXX=g++-14
export SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/
```

> The first two lines set GCC as default compiler in CMake. The last export is also necessary due to issues with GCC 14 and

Reviewer: ```suggestion
The first two lines set GCC as the default compiler in CMake. The last export is also necessary due to issues with GCC 14 and
```
